state-transition/core/validation_deposits.go

```diff
@@ -59,19 +60,21 @@ func (sp *StateProcessor) validateGenesisDeposits(
 	// BeaconKit enforces a cap on the validator set size.
 	// If genesis deposits breaches the cap we return an error.
 	//#nosec:G701 // can't overflow.
-	if uint64(len(deposits)) > sp.cs.ValidatorSetCap() {
+	if uint64(len(deposits)) > validatorSetCap {
 		return errors.Wrapf(
 			ErrValSetCapExceeded,
 			"validator set cap %d, deposits count %d",
-			sp.cs.ValidatorSetCap(), len(deposits),
+			validatorSetCap, len(deposits),
 		)
 	}
 	return nil
 }
 
-func (sp *StateProcessor) validateNonGenesisDeposits(
+func ValidateNonGenesisDeposits(
 	ctx context.Context,
 	st *statedb.StateDB,
+	depositStore *depositdb.KVStore,
+	maxDepositsPerBlock uint64,
```

Comment:
nit: consider passing chainSpec here instead?

Reply:
2 things:
- We don't always want to use the value from the chain spec for `maxDepositsPerBlock`. In CLI it is 0, since we are not validating a new block but the snapshotted state. In normal processing we want to use the chain spec value.
- I prefer these functions as helper/utils to take in raw values as much as possible, similarly to how `validateGenesisDeposits` takes in the validatorSetCap and not the chainSpec. Then these functions can be used in different contexts easily.